services/director-v2/src/simcore_service_director_v2/modules/db/repositories/groups_extra_properties.py

```diff
@@ -1,28 +1,33 @@
-import sqlalchemy as sa
-from models_library.products import ProductName
-from models_library.users import UserID
+from simcore_postgres_database.utils_groups_extra_properties import (
+    GroupExtraProperties,
+    GroupExtraPropertiesRepo,
+)
 
-from ..tables import groups_extra_properties, user_to_groups
 from ._base import BaseRepository
 
 
 class GroupsExtraPropertiesRepository(BaseRepository):
-    async def has_internet_access(
-        self, user_id: UserID, product_name: ProductName
-    ) -> bool:
+    async def _get_aggregated_properties_for_user(
+        self,
+        *,
+        user_id: int,
+        product_name: str,
+    ) -> GroupExtraProperties:
         async with self.db_engine.acquire() as conn:
-            # checks if one of the groups which the user is part of has internet access
-            select_stmt = sa.select(
-                [groups_extra_properties.c.internet_access]
-            ).select_from(
-                user_to_groups.join(
-                    groups_extra_properties,
-                    (groups_extra_properties.c.group_id == user_to_groups.c.gid)
-                    & (user_to_groups.c.uid == user_id)
-                    & (groups_extra_properties.c.internet_access == True)
-                    & (groups_extra_properties.c.product_name == product_name),
-                )
+            return await GroupExtraPropertiesRepo.get_aggregated_properties_for_user(
+                conn, user_id=user_id, product_name=product_name
             )
 
-            user_with_access = await conn.scalar(select_stmt)
-            return user_with_access is not None
+    async def has_internet_access(self, *, user_id: int, product_name: str) -> bool:
+        group_extra_properties = await self._get_aggregated_properties_for_user(
```

Comment:
MINOR: why don't yo ujust query this flag instead of getting al the properties? You are a the db interface right now. E.g. can pass the select as option?

Reply:
This is a bit of special table. We first aggregate all groups for the user and then we can conclude if the property is available. That's why it's not a direct query.

Have a look a the function that is called for more info:

https://github.com/ITISFoundation/osparc-simcore/blob/master/packages/postgres-database/src/simcore_postgres_database/utils_groups_extra_properties.py#L112-L153